crail/src/main/java/com/yahoo/ycsb/db/CrailClient.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) 2015 YCSB contributors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package com.yahoo.ycsb.db;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.Vector;
+
+import org.apache.crail.CrailBufferedInputStream;
+import org.apache.crail.CrailBufferedOutputStream;
+import org.apache.crail.CrailStore;
+import org.apache.crail.CrailKeyValue;
+import org.apache.crail.CrailLocationClass;
+import org.apache.crail.CrailNodeType;
+import org.apache.crail.CrailStorageClass;
+import org.apache.crail.conf.CrailConfiguration;
+import com.yahoo.ycsb.ByteArrayByteIterator;
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.DB;
+import com.yahoo.ycsb.DBException;
+import com.yahoo.ycsb.Status;
+
+/**
+ * Crail binding for <a href="http://crail.apache.org/">Crail</a>.
+ */
+public class CrailClient extends DB {
+  private CrailStore client;
+  private long startTime;
+  private long endTime;
+  private String usertable;
+  
+  @Override
+  public void init() throws DBException {
+    super.init();
+    try {
+      CrailConfiguration crailConf = new CrailConfiguration();
+      this.client = CrailStore.newInstance(crailConf);
+      this.usertable = "usertable";
+      if (client.lookup(usertable).get() == null) {
+        client.create(usertable, CrailNodeType.TABLE, CrailStorageClass.DEFAULT, 
+          CrailLocationClass.DEFAULT, true).get().syncDir();
+      }
+      this.startTime = System.nanoTime();
+      System.out.println("YCSB/Crail client initialized");
```

Comment:
please use a logging framework instead of System.out. Preferably slf4j.

Reply:
Thanks for the comments Sean, sorry this is still due, I have time to work on this next week.